@here/harp-mapview/lib/VisibleTileSet.ts

```diff
@@ -886,76 +895,65 @@ export class VisibleTileSet {
+                : SearchDirection.NONE;
+        return { searchDirection, searchLevelsUp, searchLevelsDown };
     }
 
     /**
-     * Search cache to replace visible but yet empty tiles with already loaded siblings in nearby
-     * zoom levels.
+     * Populates the list of tiles to render, see "renderedTiles". Tiles that are loaded and which
+     * are an exact match are added straight to the list, tiles that are still loading are replaced
+     * with tiles in the cache that are either a parent or child of the requested tile. This helps
+     * to prevent flickering when zooming in / out. The distance to search is based on the options
+     * [[quadTreeSearchDistanceDown]] and [[quadTreeSearchDistanceUp]].
+     *
+     * Each [[DataSource]] can also switch this behaviour on / off using the [[canFallback]] flag.
      *
-     * Useful, when zooming in/out and when "newly elected" tiles are not yet loaded. Prevents
-     * flickering by rendering already loaded tiles from upper/higher zoom levels.
      */
-    private fillMissingTilesFromCache() {
+    private populateRenderedTiles() {
         this.dataSourceTileList.forEach(renderListEntry => {
-            const dataSource = renderListEntry.dataSource;
-            const dataZoomLevel = renderListEntry.zoomLevel;
             const renderedTiles = renderListEntry.renderedTiles;
 
-            // Direction in quad tree to search: up -> shallower levels, down -> deeper levels.
-            enum SearchDirection {
-                NONE,
-                UP,
-                DOWN,
-                BOTH
-            }
-            let defaultSearchDirection = SearchDirection.NONE;
-
-            const { searchLevelsUp, searchLevelsDown } = this.getCacheSearchLevels(
-                dataSource,
-                dataZoomLevel
-            );
-
-            defaultSearchDirection =
-                searchLevelsDown > 0 && searchLevelsUp > 0
-                    ? SearchDirection.BOTH
-                    : searchLevelsDown > 0
-                    ? SearchDirection.DOWN
-                    : searchLevelsUp > 0
-                    ? SearchDirection.UP
-                    : SearchDirection.NONE;
-
-            const incompleteTiles: Map<number, SearchDirection> = new Map();
+            // Tiles for which we need to fall(back/forward) to.
+            const incompleteTiles: number[] = [];
 
+            // Populate the list of tiles which can be shown ("renderedTiles"), and the list of
+            // tiles that are incomplete, and for which we search for an alternative
+            // ("incompleteTiles").
             renderListEntry.visibleTiles.forEach(tile => {
-                const tileCode = TileOffsetUtils.getKeyForTileKeyAndOffset(
-                    tile.tileKey,
-                    tile.offset
-                );
                 tile.levelOffset = 0;
                 if (tile.hasGeometry) {
-                    renderedTiles.set(tileCode, tile);
```

Comment:
minor optimization / code improvement by computing it only once and caching in the Tile